internal/impl/observability.go

```diff
@@ -0,0 +1,1449 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package impl
+
+import (
+	_ "embed"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/ServiceWeaver/weaver/runtime/protos"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/apps/v1"
+	_ "k8s.io/api/autoscaling/v2beta2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/yaml"
+)
+
+// TODO(rgrandl): We might want to revisit the way we integrate with external
+// systems to export observability information. For example, we might want to
+// create an agent abstraction that is used by the babysitters to export
+// observability info. Different implementations of these agents will behave
+// differently. For example, a Jaeger agent, will simply export traces to a Jaeger
+// service. A Prometheus agent will export a /metrics endpoin that can be scraped
+// by a Prometheus service. Another agent might want to convert the otel traces to
+// a different format and export it (e.g., Elastic). This way, we can add agent
+// implementations for any observability systems.
+
+const (
+	// The names of the observability services that interact with the application.
```

Comment:
Do you want to make it clear in the names and comments of these which ones apply to the services we start vs. to all services. For example:

```
// Name of the Jaeger container image used for the automatically started Jaeger service.
s/jaegerImageName/autoJaegerImageName/
```